src/main/java/org/openstreetmap/atlas/geography/atlas/change/MemberMerger.java

```diff
@@ -0,0 +1,438 @@
+            }
+        }
+        /*
+         * If there was no simple strategy, we have to fail.
+         */
+        else
+        {
+            throw new CoreException(
+                    "Conflicting members and no merge strategy for {}; afterView: {} vs {}",
+                    this.memberName, afterMemberLeft, afterMemberRight);
+        }
+
+        return new MergedMemberBean<>(beforeMemberResult, afterMemberResult);
+    }
+
+    /**
+     * Merge a member that has conflicting beforeViews. This can happen occasionally with
+     * {@link RelationBean}s and the in/out {@link Edge} identifier sets in {@link Node}, since
+     * these may be inconsistent across shards.
+     *
+     * @param beforeMemberLeft
+     *            the left side before view of the member
+     * @param afterMemberLeft
+     *            the left side after view of the member
+     * @param beforeMemberRight
+     *            the right side before view of the member
+     * @param afterMemberRight
+     *            the right side after view of the member
+     * @return a {@link MergedMemberBean} containing the merged beforeMember view and the merged
+     *         afterMember view
+     */
+    private MergedMemberBean<M> mergeMemberWithConflictingBeforeViews(final M beforeMemberLeft,
+            final M afterMemberLeft, final M beforeMemberRight, final M afterMemberRight)
+    {
+        final M beforeMemberResult;
+        final M afterMemberResult;
+
+        if (this.afterViewConflictingBeforeMerger == null)
+        {
+            throw new CoreException(
+                    "Conflicting beforeMembers and no afterViewConflictingBeforeMerger for {}; beforeView: {} vs {}",
+                    this.memberName, beforeMemberLeft, beforeMemberRight);
+        }
+
+        if (this.beforeViewMerger == null)
+        {
+            throw new CoreException(
+                    "Conflicting beforeMembers and no beforeViewMerger for {}; beforeView: {} vs {}",
+                    this.memberName, beforeMemberLeft, beforeMemberRight);
+        }
+
+        try
+        {
+            beforeMemberResult = this.beforeViewMerger.apply(beforeMemberLeft, beforeMemberRight);
+        }
+        catch (final Exception exception)
+        {
+            throw new CoreException(
+                    "Attempted beforeView merge failed for {} with beforeView: {} vs {}",
+                    this.memberName, beforeMemberLeft, beforeMemberRight, exception);
+        }
+
+        try
+        {
+            afterMemberResult = this.afterViewConflictingBeforeMerger.apply(beforeMemberLeft,
+                    afterMemberLeft, beforeMemberRight, afterMemberRight);
+        }
+        catch (final Exception exception)
+        {
+            throw new CoreException(
+                    "Attempted afterViewConflictingBeforeMerge merge failed for {} with beforeView: {} vs {}; afterView: {} vs {}",
```

Comment:
Is there a way to make this message more descriptive? It seems afterViewConflictingBeforeMerge is very implementation-specific, and when a user sees that in a stack trace, they wont understand

Reply:
I updated the message, let me know what you think. We can always change these messages in the future should we think of something better.